### PR TITLE
Remove custom code for LdagL/SteadyState gradient computation

### PR DESCRIPTION
**Pull request by PhilipVinc**
The special code for the steady state computation is quite old.
With the big operator rewrite for chunking I did not realise it immediately, but I implemented a more efficient version of squared operator gradient.

This PR simply removes the old special code and falls back to using AD on `nkjax.expect` to compute gradient of LdagL.
It's also 10% faster.

### CODE REVIEW

**Comment by PhilipVinc** [comment]
As I wrote in a comment of the tests...
A newly added test fails when run under `pytest-xdist`. I can't reproduce locally on my laptop and my workstation (though if somebody tried to run it and managed to reproduce, I would be happy to investigate more..)

I think it's a jax bug, but can't investigate further.
I disabled the test on CI, but it will run locally.
